coderd/workspaces.go

```diff
@@ -1366,7 +1471,7 @@ func (api *API) workspaceData(ctx context.Context, workspaces []database.Workspa
 	}, nil
 }
 
-func convertWorkspaces(workspaces []database.Workspace, data workspaceData) ([]codersdk.Workspace, error) {
+func convertWorkspaces(requesterID uuid.UUID, workspaces []database.Workspace, data workspaceData) ([]codersdk.Workspace, error) {
 	buildByWorkspaceID := map[uuid.UUID]codersdk.WorkspaceBuild{}
```

Comment:
What do you think about adding a precondition here to fail fast if uuid.UUID is empty/null/zero?

Reply:
What would be the purpose of this?

Reply:
To catch misuse of `convertWorkspaces` where it is called without context.

Reply:
Will address in a follow-up PR.